### PR TITLE
SECD-799 Add stat for attribution failure

### PR DESCRIPTION
**Pull request by faldridge**

### CODE REVIEW

File: pkg/handlers/v1/attribute.go [hunk 1]
```diff
@@ -20,12 +20,14 @@ type AttributeHandler struct {
...
 func (h *AttributeHandler) Handle(ctx context.Context, assetVulns domain.NexposeAssetVulnerabilities) error {
 	logger := h.LogFn(ctx)
+	stater := h.StatFn(ctx)
 
 	attributedAssetVulns, err := h.AssetAttributor.Attribute(ctx, assetVulns)
 	if err != nil {
 		switch err.(type) {
 		case domain.AssetNotFoundError:
 			logger.Error(logs.AssetNotFoundError{Reason: err.Error()})
+			stater.Count("sec.nexpose_vuln_attribution_failure", 1)
```
Comment (kconwayatlassian) [inline pkg/handlers/v1/attribute.go]: suggestion: I don't believe this metric adds value over the returned error. Returning an error will trigger a failure metric in CloudWatch as part of the standard Lambda metrics suite and will trigger all relevant 5xx metrics when using AWS API Gateway or serverfull-gateway.

Comment (bkrebsbach) [reply pkg/handlers/v1/attribute.go]: Nit: mixing `.` and `_` seems strange to me

Comment (faldridge) [reply pkg/handlers/v1/attribute.go]: Well, in our service configuration we are using serverfull-gateway to coerce the 404 response to a 204 due to some idiosyncrasies with our usage of SQS dead-letter queues.

In that scenario, does the gateway still provide the same metrics (I would specifically need a 404 metric to get the value I'm looking for here)?

Comment (faldridge) [reply pkg/handlers/v1/attribute.go]: @bkrebsbach Yeah, we really need to converge on a standard. For now, I'll remove the prefix and make it look more like our http ones. 👍

Comment (faldridge) [reply pkg/handlers/v1/attribute.go]: @bkrebsbach Changed to `nexpose.attribution.failure`.

Comment (kconwayatlassian) [reply pkg/handlers/v1/attribute.go]: @faldridge I think there are some use cases I don't understand here. For example, why are we converting a 404 to 204 when the original 404 is, itself, a synthetic status code derived by interpreting the error? It's not clear to me why this condition is an error in code but not considered an error at runtime. It's also not clear why this dead-letter strategy is affecting the operational behavior for all deployments given that it is a detail of a specific deployment that uses dead-letter queues.

I'm not going to block the PR on my suggestion but I suspect there are other things we might look at changing with our internal deployment of this system that would achieve the desired effect without changing the core project.

Comment (faldridge) [reply pkg/handlers/v1/attribute.go]: This is a conversation we do need to have and soon, but for this PR, after a conversation with @LorenBoren, I realized we can indeed use existing httpstats metrics to get the visibility into attribution failures I was looking for. I will decline this PR.

**Pull request closed without merging**